controllers/chat/sendSignedMessage.js

```diff
@@ -0,0 +1,79 @@
+const Validator = require('fastest-validator');
+const {StreamChat} = require('stream-chat');
+const {responseError, responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+
+const v = new Validator();
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const sendSignedMesage = async (req, res) => {
+  const schema = {
+    channelId: 'string|empty:false',
+    message: 'string|empty:false',
+    channelType: {
+      type: 'enum',
+      values: Object.values(CHANNEL_TYPE),
+      empty: false
+    }
+  };
+  const validated = v.validate(req.body, schema);
+  if (validated.length)
+    return res.status(403).json({
+      message: 'Error validation',
+      error: validated
+    });
+
+  const {channelId, message, channelType} = req.body;
+
+  let channelTypeDef;
+
+  switch (channelType) {
+    case CHANNEL_TYPE.CHAT:
+    case CHANNEL_TYPE.ANONYMOUS:
+      channelTypeDef = 'messaging';
+      break;
+
+    case CHANNEL_TYPE.GROUP:
+      channelTypeDef = 'group';
+      break;
+
+    default:
+      return res.status(403).json({
+        message: 'Error validation',
+        error: 'Channel type not found'
+      });
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The switch statement for determining the `channelTypeDef` based on `channelType` is clear and understandable. However, consider using a mapping object instead of a switch statement for improved readability and performance.

```diff
-  let channelTypeDef;
-  
-  switch (channelType) {
-    case CHANNEL_TYPE.CHAT:
-    case CHANNEL_TYPE.ANONYMOUS:
-      channelTypeDef = 'messaging';
-      break;
-
-    case CHANNEL_TYPE.GROUP:
-      channelTypeDef = 'group';
-      break;
-
-    default:
-      return res.status(403).json({
-        message: 'Error validation',
-        error: 'Channel type not found'
-      });
-  }
+  const channelTypeMapping = {
+    [CHANNEL_TYPE.CHAT]: 'messaging',
+    [CHANNEL_TYPE.ANONYMOUS]: 'messaging',
+    [CHANNEL_TYPE.GROUP]: 'group'
+  };
+
+  const channelTypeDef = channelTypeMapping[channelType];
+
+  if (!channelTypeDef) {
+    return res.status(403).json({
+      message: 'Error validation',
+      error: 'Channel type not found'
+    });
+  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->